Makefile

```diff
@@ -122,6 +122,41 @@ ENVOY_FILE ?= envoy-bin/envoy-static-stripped
   BASE_PY_IMAGE    ?= $(BASE_DOCKER_REPO):py-$(BASE_PY_RELVER)
 # END LIST OF VARIABLES REQUIRING `make docker-update-base`.
 
+#### Test service Dockerfile stuff.
+# The test services live in subdirectories of test-services. TEST_SERVICE_ROOTS
+# is the list of these directories.
+
+TEST_SERVICE_ROOTS = $(notdir $(wildcard test-services/*))
+
+# TEST_SERVICE_IMAGES maps each TEST_SERVICE_ROOT to test-$root.docker, since
+# those are the names of the individual targets. We also add the auth-tls
+# target here, by hand -- it has a special rule since it's also built from the
+# test-services/auth directory.
+TEST_SERVICE_IMAGES = $(patsubst %,test-%.docker,$(TEST_SERVICE_ROOTS) auth-tls)
+
+# Set default tag values...
+docker.tag.release = $(AMBASSADOR_DOCKER_TAG)
+docker.tag.local = $(AMBASSADOR_DOCKER_TAG)
+
+TEST_SERVICE_VERSION = 0.0.2
+
+# ...then set overrides for the test services.
+test-%.docker.tag.release: docker.tag.release = quay.io/datawire/test_services:$(notdir $*)-$(TEST_SERVICE_VERSION)
+
+LOCAL_REPO = $(if $(filter-out -,$(DOCKER_REGISTRY)),$(DOCKER_REGISTRY)/test_services,test_services)
+test-%.docker.tag.local: docker.tag.local = $(LOCAL_REPO):$(notdir $*)-$(GIT_DESCRIPTION)
```

Comment:
Is `.tag.local` used for things other than the test services? Why not just write
```Makefile
docker.tag.local = $(LOCAL_REPO):$(notdir $*)-$(GIT_DESCRIPTION)
```
instead of doing an override?

Reply:
I _think_ it's going to get used for other things shortly.